File: src/java/magma/app/compile/type/PrimitiveNode.java
package magma.app.compile.type;

import magma.app.TypeCompiler;
import magma.app.compile.node.Node;

public enum PrimitiveNode implements Node {
    String("string"),
    Number("number"),
    Boolean("boolean"),
    Var("var"),
    Void("void"),
    Unknown("unknown");

    private final String value;

    PrimitiveNode(String value) {
        this.value = value;
    }

    public String generateNode() {
        return this.value;
    }

    public boolean isVar() {
        return PrimitiveNode.Var == this;
    }

    public String generateBeforeName() {
        return "";
    }

    public String generateSimple() {
        return TypeCompiler.generateNode(this);
    }

    public boolean is(String type) {
        return type.equals(this.name().toLowerCase());
    }
}

File: src/ts/magma/app/compile/type/PrimitiveNode.ts
import { Node } from "../../../../magma/app/compile/node/Node";
import { TypeCompiler } from "../../../../magma/app/TypeCompiler";
export class PrimitiveNode implements Node {
	static String: PrimitiveNode = new PrimitiveNode("string");
	static Number: PrimitiveNode = new PrimitiveNode("number");
	static Boolean: PrimitiveNode = new PrimitiveNode("boolean");
	static Var: PrimitiveNode = new PrimitiveNode("var");
	static Void: PrimitiveNode = new PrimitiveNode("void");
	static Unknown: PrimitiveNode = new PrimitiveNode("unknown");
	value: string;
	constructor (value: string) {
		this.value/*unknown*/ = value/*string*/;
	}
	generateNode(): string {
		return this.value/*unknown*/;
	}
	isVar(): boolean {
		return PrimitiveNode.Var === this/*unknown*/;
	}
	generateBeforeName(): string {
		return ""/*unknown*/;
	}
	generateSimple(): string {
		return TypeCompiler.generateNode(this)/*unknown*/;
	}
	is(type: string): boolean {
		return type.equals(this.name().toLowerCase())/*unknown*/;
	}
}
